src/command/command.rs

```diff
@@ -64,6 +66,35 @@ impl Command {
                 }
                 Ok((Command::Command, 2))
             }
+            _ if cmd == "GET" => {
```

Comment:
Very messy and out of scope of this PR for changing it in a significant way.

Reply:
`_ if cmd == "GET"` can be just `"GET"`

A literal can be a `match` pattern: https://doc.rust-lang.org/stable/reference/patterns.html#literal-patterns

Reply:
If you did that because of borrow checker issues, try:
```rust
match &cmd
```

on line 51.

Reply:
fixed in 31b3065